### ISSUE TITLE
🐛 BUG: [Renderers] Support importing Node exports during SSR

### ISSUE THREAD

**Issue by eyelidlessness** [snowpackjs/astro#552]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Currently Astro SSR imports from a package’s `exports.browser` rather than its `module` or otherwise appropriate Node resolution. Attempting to import the Node module directly fails with an error (sorry for not including the exact error in this report, reporting from mobile as discussed on Discord).

This has been an issue as I’ve tried to prototype/build a renderer for Solid, because its `renderToString` implementation, under `solid-js/web`, is a no-op in the `browser` build. For reference, this is the package’s current exports: https://github.com/solidjs/solid/blob/0e30ba438d24729d94fae34f6c6ec6d3dd626515/packages/solid/web/package.json#L3-L8

### Expected Behavior

In SSR a renderer should access Node modules (presumably by default but at least be allowed).

### Steps to reproduce

```bash
1. Install solid-js in a renderer
2. In the renderer’s server module, `import 'solid-js/web/dist/server.js'`

(Alternatively `import 'solid-js/web'`, which I’d expect to resolve to the Node export; I’m reporting this pragmatically in hopes of supporting quick resolution for a Solid renderer.)
```

**Comment by matthewp** [comment]
Thanks @eyelidlessness, I suppose this means you need a different build for ssr and the browser, right?

**Comment by eyelidlessness** [comment]
> Thanks @eyelidlessness, I suppose this means you need a different build for ssr and the browser, right?

In this case yes, but I think that may be orthogonal. What I need is for a `renderer.server` to be able to `import 'solid-js/web/dist/server.js'`. Currently that is impossible because imports use browser resolution.

**Comment by natemoo-re** [comment]
Yeah this is an upstream issue with Snowpack. There's currently no way to override `esinstall` to use something other than the `browser` entrypoint if one exists.

See also #459. cc @drwpow

**Comment by eyelidlessness** [comment]
@FredKSchott I’m not sure why you removed the bug label. It’s definitely expected that SSR operates in an server environment and has access to server modules.

Is this being treated as a feature request? As far as I’m aware this will affect any SSR functionality that differs from its client counterpart. I understand the workaround might be to use module aliases (which I hope to validate when I get bandwidth), but that’s quite surprising for renderer development and pretty much a non-starter for Astro users with a Snowpack config.

**Comment by matthewp** [comment]
He removed all the bug labels and made a bug project instead.

**Comment by eyelidlessness** [comment]
If anyone knows who to @ at GH this feels like something that should be possible to signal in issue changes.

**Comment by matthewp** [comment]
Just coming back to this, the issue is that snowpack does not build separate SSR builds for packages. If we did that we could use the correct export map field.

**Comment by natemoo-re** [comment]
Confirming that this is fixed in `astro@next`!

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
